### PR TITLE
WebGPU: Add WebGPUUniformBuffer

### PR DESCRIPTION
**Pull request by sunag**
**Description**

This will simplify the creation of `UBO` in new `NodeMaterial` system and can be useful for coded shader.

### CODE REVIEW

File: examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	}
+
+	getBuffer() {
+
+		return this.buffer;
+
+	}
+
+	update() {
```
Comment (Mugen87) [inline examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js:45]: What is this method for?

File: examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		this.visibility = GPUShaderStage.VERTEX;
+
+		this.usage = GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST;
+
+		this.buffer = buffer;
+		this.bufferGPU = null; // set by the renderer
+
+	}
+
+	setOnBeforeUpdate( callback ) {
```
Comment (Mugen87) [inline examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js:23]: Do we still need this method? `onBeforeUpdate()` is still called in `WebGPUBindings` however it seems we never assign a callback.

File: examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js (hunk 1 above)
Comment (Mugen87) [inline examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js:45]: What is the purpose of this method?

File: examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js (hunk 2 above)
Comment (sunag) [reply examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js]: Yeah, I will remove.

File: examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/renderers/webgpu/WebGPUUniformBuffer.js:35]: >  an interface method.

Okay, that cleared thing up!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**